src/sync/background/post-receive-processor.ts

```diff
@@ -0,0 +1,50 @@
+import { SyncPostReceiveProcessor } from '@worldbrain/storex-sync'
+import { COLLECTION_NAMES as PAGES_COLLECTION_NAMES } from '@worldbrain/memex-storage/lib/pages/constants'
+
+import { FetchPageDataProcessor } from 'src/page-analysis/background/fetch-page-data-processor'
+import { SharedSyncLogEntry } from '@worldbrain/storex-sync/lib/shared-sync-log/types'
+import { PageFetchBacklogBackground } from 'src/page-fetch-backlog/background'
+
+export class PostReceiveProcessor {
+    constructor(
+        private props: {
+            fetchPageData: FetchPageDataProcessor
+            pageFetchBacklog?: PageFetchBacklogBackground
+        },
+    ) {}
+
+    private async handleFailure(
+        entry: SharedSyncLogEntry<'deserialized-data'>,
+    ) {
+        if (this.props.pageFetchBacklog) {
+            await this.props.pageFetchBacklog.enqueueEntry({
+                url: entry.data.pk,
+            })
+        }
+    }
+
+    processor: SyncPostReceiveProcessor = async ({ entry, ...params }) => {
+        if (
+            entry.data.collection === PAGES_COLLECTION_NAMES.page &&
+            entry.data.operation === 'create'
+        ) {
+            try {
+                const value = await this.props.fetchPageData.process(
```

Comment:
This doesn't account for ext to ext sync yet where this data is already filled in. It should not fetch page data if the data is already present.

Reply:
Should be addressed in 1ecc1af423c530ecaa623505250394782348608a